src/main/java/com/hivemq/extensions/prometheus/export/PrometheusServer.java

```diff
@@ -60,45 +64,48 @@ public PrometheusServer(
         final QueuedThreadPool queuedThreadPool = new QueuedThreadPool();
         queuedThreadPool.setMinThreads(MIN_THREADS);
         queuedThreadPool.setMaxThreads(MAX_THREADS);
-        server = new Server(queuedThreadPool);
+        this.server = new Server(queuedThreadPool);
         final ServerConnector connector = new ServerConnector(server);
         connector.setHost(configuration.hostIp());
         connector.setPort(configuration.port());
         server.setConnectors(new Connector[]{connector});
     }
 
-    public void start() {
-        dropwizardExports = new DropwizardExports(metricRegistry);
+    public void start() throws Exception {
+        final DropwizardExports dropwizardExports = new DropwizardExports(metricRegistry);
         CollectorRegistry.defaultRegistry.register(dropwizardExports);
+        dropwizardExportsRef.set(dropwizardExports);
 
         final ServletContextHandler context = new ServletContextHandler();
         context.setContextPath("/");
         server.setHandler(context);
         context.addServlet(new ServletHolder(new MonitoredMetricServlet(metricRegistry)), configuration.metricPath());
-        try {
-            server.start();
-        } catch (final Exception e) {
-            log.error("Error starting the Jetty Server for Prometheus Extension");
```

Comment:
We only logged startup errors, but then proceed without properly preventing the extension startup. We now just throw all exceptions into the main method and handle them there.